### PR TITLE
Convolution operation sweep suite

### PR DESCRIPTION
**Pull request by nilaykp**
### Ticket
[Link](https://github.com/tenstorrent/tt-metal/issues/12200) to Github Issue

### Problem description
Create sweep suite for convolution using sweep framework

### What's changed
Adding new sweep suite for convolution operation.

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py
Comment (mywoodstock) [inline tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py:108]: I think we should be able to get the arch name directly from `device` object? Then we don't need to rely on the env vars.

File: tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py
Comment (mywoodstock) [inline tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py:121]: should this be called like `op_parameters` or something?

File: tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py
Comment (mywoodstock) [inline tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py:116]: i think we dont need to explicitly sync before closing the device.

File: tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py
Comment (mywoodstock) [inline tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py:182]: not needed.

**Review by mywoodstock — COMMENTED** [review]
Can we make the all the repeated code as common functions and call them from the each test file?

File: tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py]: Done.

File: tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py]: changed to input_specs.

File: tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py]: Done.

File: tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/full/conv2d_block_sharded.py]: Done.

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (mywoodstock) [inline tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py:20]: Can you add a comment line with the names of each of the params in the input_specs

Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py:22]: Done.

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py]: Made is true by default(false would increase number of tests which I think is not necessary), Please do let me know if false test cases are also needed.

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py]: Made is true by default(false would increase number of tests which I think is not necessary), Please do let me know if false test cases are also needed.

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py]: Done.

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py]: Done.

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (mywoodstock) [inline tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py:427]: Pavle is working on auto-sharding selection -- this is the PR: https://github.com/tenstorrent/tt-metal/pull/13573
May be wait for this to land in `main` and then you can remove this sharding_scheme parameter and use the auto shard -- this will reduce the #tests by a third.

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (mywoodstock) [inline tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py:435]: In favor of reduced sweep time, lets only test with bfloat8_b (both act and weights)

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (mywoodstock) [inline tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py:436]: `LoFi` instead.

File: tests/sweep_framework/sweeps/max_pool2d/short/max_pool2d_short_sweep.py
Comment (mywoodstock) [inline tests/sweep_framework/sweeps/max_pool2d/short/max_pool2d_short_sweep.py:57]: Lets use just one case -- `True`

**Review by mywoodstock — APPROVED** [review]

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py:427]: Sure.

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py]: Done.

File: tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/conv2d/short/conv2d_short_sweep.py]: Done.

File: tests/sweep_framework/sweeps/max_pool2d/short/max_pool2d_short_sweep.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/max_pool2d/short/max_pool2d_short_sweep.py]: Done.

**Comment by mywoodstock** [comment]
@nkpatel-tt the auto sharding is in `main` now, so you can remove the sharding scheme parameter and use `auto_shard=True`. Once thats done I think you are good to merge and enable on CI.

**Review by jdesousa-TT — APPROVED** [review]

**Comment by jdesousa-TT** [comment]
[ttnn - Run sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11295339861)
[ttnn - Run sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11295342298)
[ttnn - Run sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11295344311)
[ttnn - Run sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11295346347)

Running these to check for syntax/import issues, and that we can actually generate all these test vectors. It likely won't finish executing the tests because it will take too long, but let's see if we can actually even generate fully.

File: tests/sweep_framework/sweep_utils/conv2d_common.py
Comment (jdesousa-TT) [inline tests/sweep_framework/sweep_utils/conv2d_common.py:42]: This should be device.arch, I think, and the return is an enum "GRAYSKULL" or "WORMHOLE_B0". You can also use `is_wormhole_b0` / `is_grayskull` like in `tests/sweep_framework/sweeps/eltwise/unary/frac/frac.py` for example.

Right now it always gets set to `Unknown`.

File: tests/sweep_framework/sweeps/max_pool2d/short/max_pool2d_short_sweep.py
Comment (mywoodstock) [inline tests/sweep_framework/sweeps/max_pool2d/short/max_pool2d_short_sweep.py:22]: please add a comment line specifying list of parameter names.

File: tests/sweep_framework/sweep_utils/conv2d_common.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweep_utils/conv2d_common.py]: Done.

File: tests/sweep_framework/sweeps/max_pool2d/short/max_pool2d_short_sweep.py
Comment (nkpatel-tt) [reply tests/sweep_framework/sweeps/max_pool2d/short/max_pool2d_short_sweep.py:24]: Done.

**Comment by nkpatel-tt** [comment]
> Could you please clean up the commit history? IMO, this should be maybe 3-5 commits

Squashed in 2 commits.

**Comment by nkpatel-tt** [comment]
Running all added sweeps to be sure of functionality (will cancel after some time to free machine)
- [ttnn - Run sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11346453083)
- [ttnn - Run sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11346457099)
- [ttnn - Run sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11346457246)
- [ttnn - Run sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11346461164)
- [ttnn - Run sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11346464569)
- [ttnn - Run sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11346468717)

**Pull request merged**
